File: apps/web/src/lib/saudacao.ts
export type Saudacao = "Bom dia" | "Boa tarde" | "Boa noite";

export function saudacaoPorHora(h: number): Saudacao {
  if (h < 12) return "Bom dia";
  if (h < 18) return "Boa tarde";
  return "Boa noite";
}

export function saudacaoAgora(now: Date = new Date()): Saudacao {
  return saudacaoPorHora(now.getHours());
}

File: apps/agent/src/Timesheet.Agent.Ui/Common/Saudacao.cs
namespace Timesheet.Agent.Ui.Common;

public static class Saudacao
{
    /// <summary>
    /// Retorna a saudação conforme a faixa horária:
    /// 0–11 → "Bom dia", 12–17 → "Boa tarde", 18–23 → "Boa noite"
    /// </summary>
    public static string Para(int hora) => hora switch
    {
        >= 0 and <= 11 => "Bom dia",
        >= 12 and <= 17 => "Boa tarde",
        _ => "Boa noite"
    };
}
